### PR TITLE
Consume upstream `quickcheck` implementation for `Signature` and `SignatureType`

### PR DESCRIPTION
**Pull request by jdjaustin**
## Summary of changes

Changes introduced in this pull request:
- Remove custom quickcheck implementation for `Signature` and `SignatureType` and instead use the implementation in `fvm_shared`.

## Reference issue to close (if applicable)

Closes respective part of associated task in #2143 

## Other information and links

N/A

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

File: utils/json/src/signature.rs [hunk 1]
```diff
@@ -119,46 +119,22 @@ mod tests {
...
+        assert_eq!(signature, parsed);
     }
 
     #[quickcheck]
-    fn signaturetype_roundtrip(sigtype: SignatureTypeWrapper) {
-        let serialized = serde_json::to_string(&SignatureTypeJson(sigtype.sigtype)).unwrap();
-        let parsed: SignatureTypeJson = serde_json::from_str(&serialized).unwrap();
-        assert_eq!(sigtype.sigtype, parsed.0);
+    fn signaturetype_roundtrip(sigtype: SignatureType) {
+        let serialized1: String =
```
Comment (lemmih) [inline utils/json/src/signature.rs:132]: Why rename `serialized` to `serialized1`?

File: utils/json/src/signature.rs [hunk 2]
```diff
@@ -119,46 +119,22 @@ mod tests {
...
-        }
-    }
-
     #[quickcheck]
-    fn signature_roundtrip(signature: SignatureWrapper) {
-        let serialized = serde_json::to_string(&SignatureJsonRef(&signature.signature)).unwrap();
-        let parsed: SignatureJson = serde_json::from_str(&serialized).unwrap();
-        assert_eq!(signature.signature, parsed.0);
+    fn signature_roundtrip(signature: Signature) {
+        let serialized: String =
```
Comment (lemmih) [inline utils/json/src/signature.rs:124]: Is the `: String` type signature necessary?

File: utils/json/src/signature.rs (hunk 1 above)
Comment (jdjaustin) [reply utils/json/src/signature.rs:132]: Ah that was just part of my troubleshooting before I discovered I needed to rename `super::json::serialize` to `super::json::signature_type::serialize` (and deserialize respectively). I'll fix that.

File: utils/json/src/signature.rs (hunk 2 above)
Comment (jdjaustin) [reply utils/json/src/signature.rs:124]: I know we've been using the explicit type signature in other tests, but apparently we don't need it here!

**Review by hanabi1224 — APPROVED** [review]

File: utils/json/src/signature.rs [hunk 3]
```diff
@@ -119,46 +119,21 @@ mod tests {
...
-    }
-
     #[quickcheck]
-    fn signature_roundtrip(signature: SignatureWrapper) {
-        let serialized = serde_json::to_string(&SignatureJsonRef(&signature.signature)).unwrap();
-        let parsed: SignatureJson = serde_json::from_str(&serialized).unwrap();
-        assert_eq!(signature.signature, parsed.0);
+    fn signature_roundtrip(signature: Signature) {
+        let serialized = forest_test_utils::to_string_with!(&signature, super::json::serialize);
+        let parsed = forest_test_utils::from_str_with!(&serialized, super::json::deserialize);
```
Comment (lemmih) [inline utils/json/src/signature.rs:125]: Why switch to `to_string_with!` rather than using `SignatureJsonRef`?

Comment (jdjaustin) [reply utils/json/src/signature.rs:125]: I thought we had preferred to use the `forest_test_utils` macros in previous `quickcheck` tests, but perhaps I'm mistaken. I would argue that it improves readability a bit, but I can change it back to using `SignatureJsonRef` if that's better.

Comment (lemmih) [reply utils/json/src/signature.rs:125]: The macros are there to assist if you do not have a helper structure such as `SignatureJsonRef`.

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**